botocore/credentials.py

```diff
@@ -1207,12 +1207,14 @@ def __init__(self, load_config, client_creator, cache, profile_name,
 
     def load(self):
         self._loaded_config = self._load_config()
-        if self._has_assume_role_config_vars():
+        profiles = self._loaded_config.get('profiles', {})
+        profile = profiles.get(self._profile_name, {})
+        if self._has_assume_role_config_vars(profile):
             return self._load_creds_via_assume_role(self._profile_name)
 
-    def _has_assume_role_config_vars(self):
-        profiles = self._loaded_config.get('profiles', {})
-        return self.ROLE_CONFIG_VAR in profiles.get(self._profile_name, {})
+    def _has_assume_role_config_vars(self, profile):
+        assume_role_keys = [self.ROLE_CONFIG_VAR]
+        return any(key in profile for key in assume_role_keys)
```

Comment:
What is the point of this?

Reply:
Just to check if the profile has a `role_arn`. I based it on the [_has_static_credentials](https://github.com/boto/botocore/blob/develop/botocore/credentials.py#L1348-L1350) method. Originally I had this as another method but merged it with `_has_assume_role_config_vars` which is plural so I left it open for extension by leaving it as a list of keys).